base/pkg/dir.jl

```diff
@@ -35,15 +35,20 @@ function init(meta::AbstractString=DEFAULT_META, branch::AbstractString=META_BRA
     end
     dir = path()
     info("Initializing package repository $dir")
+    temp_dir = joinpath(dir, "TEMP")
     metadata_dir = joinpath(dir, "METADATA")
+    #Check if TEMP dir exists from previous failed init
+    if isdir(temp_dir)
```

Comment:
I will remove that and use `Base.mktempdir()`
There is a potential for a failed `init()` to cause the subsequent tests to fail because of an unclean directory - `Pkg.status()` reported the `TEMP` folder which would cause the test `@test isempty(Pkg.installed())` to fail.